cmd/sovereignnode/incomingHeader/incomingEventsProcessor.go

```diff
@@ -25,32 +23,43 @@ const (
 	topicIDDeposit          = "deposit"
 )
 
+type confirmedBridgeOp struct {
+	hashOfHashes []byte
+	hash         []byte
+}
+
 type scrInfo struct {
 	scr  *smartContractResult.SmartContractResult
 	hash []byte
 }
 
+type eventsResult struct {
+	scrs               []*scrInfo
+	confirmedBridgeOps []*confirmedBridgeOp
+}
+
 type incomingEventsProcessor struct {
-	txPool     TransactionPool
 	marshaller marshal.Marshalizer
 	hasher     hashing.Hasher
-	pool       block.OutGoingOperationsPool
 }
 
-func (iep *incomingEventsProcessor) processIncomingEvents(events []data.EventHandler) ([]*scrInfo, error) {
+func (iep *incomingEventsProcessor) processIncomingEvents(events []data.EventHandler) (*eventsResult, error) {
 	scrs := make([]*scrInfo, 0, len(events))
+	confirmedBridgeOps := make([]*confirmedBridgeOp, 0, len(events))
 
 	for idx, event := range events {
 		topics := event.GetTopics()
 
 		var scr *scrInfo
+		var confirmedOp *confirmedBridgeOp
 		var err error
 		switch string(event.GetIdentifier()) {
 		case topicIDDeposit:
 			scr, err = iep.createSCRInfo(topics, event)
```

Comment:
treat error.

Reply:
Both errors are treated on L66 below, which unfortunately does not show up in github differences, only if you expand the window:

```go
		if err != nil {
			return nil, fmt.Errorf("%w, event idx = %d", err, idx)
		}
```